src/model/transaction/insertFragmentIntoContentState.js

```diff
@@ -100,10 +100,10 @@ function insertFragmentIntoContentState(
       var modifiedHead = block.merge({
         text: headText + appendToHead.getText(),
         characterList: headCharacters.concat(appendToHead.getCharacterList()),
+        type: (!text.length || targetOffset === 0) ? appendToHead.getType() : block.getType(),
```

Comment:
@spicyj That will change the insertion behavior. If we only check for `headText`, then we will always keep the type of the existing block, as long as that block contains text. If we insert at offset 0 though, I think that we should use the type of the new block instead of the existing one.